src/ydata_quality/drift/engine.py

```diff
@@ -106,25 +107,26 @@ def sample(self, sample: DataFrame):
         self._sample = sample
 
     @property
-    def model(self) -> Optional[ModelWrapper]:
+    def model(self) -> Optional[Union[ModelWrapper, Callable]]:
         """Returns a wrapper for the user's custom model in case the provided model was successfully tested.
         Returns False if a passed model failed the test.
         Returns None if no model was passed."""
         return self._model
 
     @model.setter
-    def model(self, model: Callable):
+    def model(self, model: Optional[Union[Callable, ModelWrapper]]):
         if model:
             if isinstance(model, ModelWrapper):
-                self._model = model
+                self._model: ModelWrapper = model
```

Comment:
🤔  i don't think this does something expected

Reply:
how come? it is already defined as a ModelWrapper (if-sentence) and maybe it can help linting tools

Reply:
because it's not a context variable, it's a variable from the class instance and you already did a `isinstance` of `ModelWrapper`, so it is for sure a `ModelWrapper`.

anyway, that `: ModelWrapper` doesn't do anything on this case. You have to declare the variable in the class context

```python
class DriftAnalyser(QualityEngine):
  _model: ModelWrapper

  __init__(...
```

perhaps that way, the linter knows that it is a Callable 🤔 
have you tried that?

Reply:
yeah, we can remove the type hint because of the if-clause.

btw, the not-callable warning was raised for the model property and not the _model, we tried a couple of solutions :hammer_and_pick: but none resolved the `self.model is not callable` pylint warning so we suppressed those manually

Reply:
removed in https://github.com/ydataai/ydata-quality/pull/46/commits/4eab0c7a658926995c2e089c4c58c36ed65c9b3c